Assets/Scripts/Combat/MusicManager.cs

```diff
@@ -14,34 +15,49 @@ public class MusicManager : MonoBehaviour
 
     void Start()
     {
+        aud[0] = gameObject.AddComponent<AudioSource>();
+        aud[1] = gameObject.AddComponent<AudioSource>();
         song = Random.Range(0, CombatSongs.Length);
     }
 
     public void SetVictory()
     {
         if (gameOver) return;
-        AudioSource thisSource = GetComponent<AudioSource>();
-        thisSource.clip = VictorySong;
-        thisSource.Play();
+        StartCoroutine(crossFade(1.0f, VictorySong));
         gameOver = true;
     }
 
     public void SetDefeat()
     {
         if (gameOver) return;
-        AudioSource thisSource = GetComponent<AudioSource>();
-        thisSource.clip = DefeatSong;
-        thisSource.Play();
+        StartCoroutine(crossFade(1.0f, DefeatSong));
         gameOver = true;
     }
 
+    IEnumerator crossFade(float duration, AudioClip fadeTo)
+    {
+        aud[1].clip = fadeTo;
+        aud[1].volume = 0f;
+        aud[1].Play();
+        float elapsed = 0.0f;
+        while (elapsed < duration)
+        {
+            aud[0].volume = 1f - (elapsed / duration);
+            aud[1].volume = elapsed / duration;
+            elapsed += Time.deltaTime;
+            yield return null;
+        }
+        aud[1].volume = 1f;
+        aud[0].volume = 0f;
+        yield break;
```

Comment:
Before exiting the crossFade method doesn't the aud[0] need to be stopped?

Reply:
Yeah, that's a better way to do it for sure.